### PR TITLE
Refactor and cleanup objects.h

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: hpcgap/src/objfgelm.h [hunk 1]
```diff
@@ -127,7 +127,7 @@ static inline Obj NewWord(Obj type, UInt npairs) {
   Obj word;
   word = NewBag(T_DATOBJ,2*sizeof(Obj)+npairs*BITS_WORDTYPE(type)/8L);
   (ADDR_OBJ(word)[1] = INTOBJ_INT(npairs));
-  SET_TYPE_DATOBJ( word, type);
+  SetTypeDatObj( word, type);
   return word;
```
Comment (markuspf) [inline hpcgap/src/objfgelm.h]: formatting is off here.

File: src/objfgelm.h [hunk 2]
```diff
@@ -127,7 +127,7 @@ static inline Obj NewWord(Obj type, UInt npairs) {
   Obj word;
   word = NewBag(T_DATOBJ,2*sizeof(Obj)+npairs*BITS_WORDTYPE(type)/8L);
   (ADDR_OBJ(word)[1] = INTOBJ_INT(npairs));
-  SET_TYPE_DATOBJ( word, type);
+  SetTypeDatObj( word, type);
   return word;
```
Comment (markuspf) [inline src/objfgelm.h]: same formatting problem here

File: src/objects.h [hunk 3]
```diff
@@ -808,9 +801,11 @@ extern void (* PrintPathFuncs[LAST_REAL_TNUM+1]) (
 
 /****************************************************************************
 **
-*F  SET_TYPE_DATOBJ( <obj>, <val> )  . . . . .  set the type of a data object
+*F  SetTypeDatobj( <obj>, <kind> ) . . . . . .  set the type of a data object
+**
+**  'SetTypeDatobj' sets the kind <kind> of the data object <obj>.
 */
-#define SET_TYPE_DATOBJ(obj,val)  (ADDR_OBJ(obj)[0] = (val))
+#define SetTypeDatObj(obj, type)  (ADDR_OBJ(obj)[0] = (type))
```
Comment (markuspf) [inline src/objects.h]: This might not be the correct time yet, but @ChrisJefferson wanted to turn these into inline functions as far as I recall.

Comment (fingolfin) [inline src/objects.h]: Well, I forced myself to refrain from such things, as that mixes syncing with changing the code -- and any change can introduce regressions. So I'd prefer to do that in a second step.

File: hpcgap/src/objfgelm.h (hunk 1 above)
Comment (fingolfin) [inline hpcgap/src/objfgelm.h]: Will adjust

**Pull request merged**